packages/plugin-char/Char.ts

```diff
@@ -53,82 +32,28 @@ export class Char extends JWPlugin {
      * If the selection is not collapsed, replace the text with the formating
      * that was present in the selection.
      *
-     * @param text
+     * @param params
      */
     insertText(params: InsertTextParams): void {
         const range = params.range || this.editor.vDocument.selection.range;
         const text = params.text;
-        const format = this._getCurrentFormat();
+        const inlinePlugin = this.editor.plugins.find(plugin => plugin instanceof Inline) as Inline; // todo: some form of hook...
+        let formats = [];
+        if (inlinePlugin) {
+            formats = inlinePlugin.getCurrentFormats();
+        }
```

Comment:
Move the `getCurrentFormats` and `formatCache` logic into `Char` since it is actually very specific to text insertion. You can even call it something like `insertTextFormat` or something similar.